scipy/stats/_continuous_distns.py

```diff
@@ -2900,6 +2900,129 @@ def _entropy(self, c):
 genhalflogistic = genhalflogistic_gen(a=0.0, name='genhalflogistic')
 
 
+class genhyperbolic_gen(rv_continuous):
+    r"""A generalized hyperbolic continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `genhyperbolic` is:
+
+    .. math::
+
+        f(x, \lambda, \alpha, \beta, \delta, \mu) = 
+           \frac{(\gamma/\delta)^\lambda}{\sqrt{2\pi}K_\lambda(\delta \gamma)}
+           e^{\beta (x - \mu)} \times \frac{K_{\lambda - 1/2}
+           (\alpha \sqrt{\delta^2 + (x - \mu)^2})}
+           {(\sqrt{\delta^2 + (x - \mu)^2} / \alpha)^{1/2 - \lambda}}
+
+    for :math:`x \in (-\infty; \infty)`,
+    and :math:`\lambda, \mu \in \mathbb{R}`
+    ans :math: `0 \le |\beta| < \alpha ` \beta,
+    and :math: `\delta \in \mathbb{R}_{>0}`.
+    
+    Where: :math:`\gamma := \sqrt{\alpha^2-\beta^2}`.
+    :math: `K_{\lambda}(.)` denotes the modified Bessel function of the third
+    kind and order :math: `\lambda` (`scipy.special.kn`)
+
+    `genhyperbolic` takes ``lmbda`` as a tail parameter for :math:`\lambda`.
+    takes ``alpha`` as a shape parameter for :math:`\alpha`.
+    takes ``beta`` as a skeweness parameter for :math:`\beta`.
+    takes ``delta`` as a scale parameter for :math:`\delta`.
+    takes ``mu`` as a location parameter for :math:`\mu`.
+
```

Comment:
It would probably be helpful to mention here all the special cases, and add links to the classes for those distributions in a ["See Also"](https://numpydoc.readthedocs.io/en/latest/format.html) section.